presto-druid/src/main/java/com/facebook/presto/druid/DruidPlanOptimizer.java

```diff
@@ -81,8 +82,13 @@ public PlanNode optimize(PlanNode maxSubplan,
             VariableAllocator variableAllocator,
             PlanNodeIdAllocator idAllocator)
     {
-        Map<PlanNodeId, TableScanNode> scanNodes = maxSubplan.accept(new TableFindingVisitor(), null);
-        return maxSubplan.accept(new Visitor(scanNodes, session, idAllocator), null);
+        if (!isComputePushdownEnabled(session)) {
```

Comment:
I had to put this here, it seems the optimize call is intended to optimize for broker queries since it changes the plan. If I don't check it here, then if it falls to the segment approach it will attempt to grab columns with the wrong column name.

```
select count(name) as count from table
```
This will attempt to load the a column called "count" from the segment, which fails.